extensions/ql-vscode/src/remote-queries/remote-queries-markdown-generation.ts

```diff
@@ -97,3 +135,39 @@ export function createMarkdownRemoteFileRef(
   const markdownLink = `[${linkText || fileLink.filePath}](${createRemoteFileRef(fileLink, startLine, endLine)})`;
   return markdownLink;
 }
+
+/**
+ * Builds an expandable markdown section of the form:
+ * <details> 
+ * <summary>title</summary>
+ * 
+ * contents
+ * 
+ * </details>
+ */
+function buildExpandableMarkdownSection(title: string, contents: MarkdownFile): MarkdownFile {
+  const expandableLines: MarkdownFile = [];
+  expandableLines.push(
+    '<details>',
+    `<summary>${title}</summary>`,
+    '',
+  );
+  expandableLines.push(...contents);
+  expandableLines.push(
+    '',
+    '</details>',
+    ''
+  );
+  return expandableLines;
+}
+
+/**
+ * Creates anchor link to a file in the gist. This is of the form:
+ * '#file-<name>-<file-extension>'
+ * 
+ * TODO: Make sure these names align with the actual file names once we upload them to a gist.
```

Comment:
This is a very good point. We currently store the contents of the Markdown files but we don't keep track of the names of these files. I think this will have to change. E.g. instead of having `type MarkdownFile = string[];` we'll need to have something line:

```ts
interface MarkdownFile {
  name: string;
  content: string[]
}
```

Anyway, something for a future PR!